addons/website_sale/static/src/js/website_sale.js

```diff
@@ -857,5 +872,77 @@ publicWidget.registry.websiteSaleCarouselProduct = publicWidget.Widget.extend({
+    // Handlers
+    //--------------------------------------------------------------------------
+
+    /**
+     * Center the selected indicator to scroll the indicators list when it
+     * overflows.
+     * 
+     * @private
+     * @param {Event} ev
+     */
+    _onSlideCarouselProduct: function (ev) {
+        const isLeftIndicators = this.$target.hasClass('o_carousel_product_left_indicators');
+        const $indicatorsDiv = isLeftIndicators ? this.$target.find('.o_carousel_product_indicators') : this.$target.find('.carousel-indicators');
+        const $indicator = $indicatorsDiv.find('[data-slide-to=' + $(ev.relatedTarget).index() + ']');
+        const indicatorsDivSize = isLeftIndicators ? $indicatorsDiv.outerHeight() : $indicatorsDiv.outerWidth();
+        const indicatorSize = isLeftIndicators ? $indicator.outerHeight() : $indicator.outerWidth();
+        const indicatorPosition = isLeftIndicators ? $indicator.position().top : $indicator.position().left;
+        const scrollSize = isLeftIndicators ? $indicatorsDiv[0].scrollHeight : $indicatorsDiv[0].scrollWidth;
+        let indicatorsPositionDiff = (indicatorPosition + (indicatorSize/2)) - (indicatorsDivSize/2);
+        indicatorsPositionDiff = Math.min(indicatorsPositionDiff, scrollSize - indicatorsDivSize);
+        const indicatorsPositionX = isLeftIndicators ? '0' : '-' + indicatorsPositionDiff;
+        const indicatorsPositionY = isLeftIndicators ? '-' + indicatorsPositionDiff : '0';
+        const translate3D = indicatorsPositionDiff > 0 ? "translate3d(" + indicatorsPositionX + "px," + indicatorsPositionY + "px,0)" : '';
+        $indicatorsDiv.css("transform", translate3D);
+    },
+    /**
+     *
+     * @private
+     * @param {Event} ev
+     */
+    _onMouseWheel: function (ev) {
+        ev.preventDefault();
+        if (ev.originalEvent.deltaY > 0) {
+            this.$target.carousel('next');
+        } else {
+            this.$target.carousel('prev');
+        }
+    },
+});
+
+publicWidget.registry.websiteSaleProductPageReviews = publicWidget.Widget.extend({
+    selector: '#o_product_page_reviews',
+    disabledInEditableMode: false,
+
+    /**
+     * @override
+     */
+    async start() {
+        await this._super(...arguments);
+        this._updateChatterComposerPosition();
+        extraMenuUpdateCallbacks.push(this._updateChatterComposerPosition.bind(this));
+    },
+    /**
+     * @override
+     */
+    destroy() {
+        this.$target.find('.o_portal_chatter_composer').css('top', '');
+        this._super(...arguments);
+    },
+
+    //--------------------------------------------------------------------------
+    // Private
+    //--------------------------------------------------------------------------
+
+    /**
+     * @private
+     */
+    _updateChatterComposerPosition() {
+        this.$target.find('.o_portal_chatter_composer').css('top', dom.scrollFixedOffset() + 20);
+    },
```

Comment:
- enable "discussion and rating" (in the customize show of the product page)
- add some comments for the product (to have a long right column)
- So now, when scrolling the comments. The text area to write a comment remain sticky to encourage the user to write a comment and to avoid white space on the left

Reply:
:+1: